blocks/library/embed/index.js

```diff
@@ -123,9 +125,14 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 							return;
 						}
 						response.json().then( ( obj ) => {
-							const { html, type, provider_name: providerName } = obj;
+							const { html, type, provider_name: providerName, code } = obj;
 							const providerNameSlug = kebabCase( toLower( providerName ) );
 
+							// invalid url, and this didn't come from submitting the form
+							if ( 'oembed_invalid_url' === code && ! event ) {
+								this.props.onReplace( createBlock( 'core/paragraph', { content: ( <a href={ url }>{ url }</a> ) } ) );
```

Comment:
Could we maybe always try to grab something from the page, even if it's just the title/URL/favicon/image/summary of the page? Saw similar suggestions on the PR, but also would make this redundant.

Reply:
I'd like to do that, and I'd like to look at it in another PR once we have the mechanism working here first. I get the feeling we'd want to give more control over what's presented for non-embeddable URLs, and this would probably go well with a custom "non-embeddable embed" block that would have options for editing the description, etc. There's some other work we need to do to make that happen, so I didn't want to delay this.

Reply:
Ah I see what you mean where you can sometimes choose the image. Maybe for now, instead of the OG title we can just display the content in the title tags (in card form)? Nothing more?

Reply:
I had a version of this that did that for URLs that didn't have OG data, but it resulted in the oembed proxy endpoint returning data  for every URL, and never saying "I can't embed this" and that caused chaos in the classic editor. That's why I ended up thinking that we need an embed block specifically for this that uses a different API and saves preview HTML as a block that the oEmbed code doesn't touch.